jsk_perception/include/jsk_perception/histogram_of_oriented_gradients.h

```diff
@@ -1,44 +1,44 @@
+// Copyright (C) 2016 by Krishneel Chaudhary @ JSK Lab, The University
+// of Tokyo
 
 #ifndef _HISTOGRAM_OF_ORIENTED_GRADIENTS_H_
 #define _HISTOGRAM_OF_ORIENTED_GRADIENTS_H_
 
-// OpenCV Header Directives
 #include <opencv2/opencv.hpp>
 #include <opencv2/highgui/highgui.hpp>
 #include <opencv2/imgproc/imgproc.hpp>
-
-#include <string>
-#include <vector>
+#include <opencv2/features2d/features2d.hpp>
 
 class HOGFeatureDescriptor {
 
-    //  HOG Configuration Params
-#define N_BINS 9
-#define ANGLE 180.0
-#define BINS_ANGLE (ANGLE / N_BINS)
-#define CELL 8
-#define BLOCK 2
-
+ protected:
+    int N_BINS;
+    int ANGLE;
+    int BINS_ANGLE;
+    int CELL;
+    int BLOCK;
+    int num_threads_;
+    
  private:
-    virtual void bilinearBinVoting(
+    virtual void histogramBinVoting(
        const float &, int &, int &);
     virtual void imageGradient(
        const cv::Mat &, cv::Mat &);
     virtual cv::Mat blockGradient(
-       const int, const int, cv::Mat &);
-    virtual cv::Mat orientationistogram(
-       const cv::Mat&, const int &, const int &, bool = false);
+        const int, const int, const int, cv::Mat &);
     virtual void getHOG(
        const cv::Mat &, cv::Mat &, cv::Mat &);
-    template<typename T>
-    T computeHOGHistogramDistances(
-       const cv::Mat &, std::vector<cv::Mat> &,
-       const int = CV_COMP_BHATTACHARYYA);
    
  public:
-    HOGFeatureDescriptor();
-    virtual cv::Mat computeHOG(
-       const cv::Mat &);
+    HOGFeatureDescriptor(
+        const int = 8, const int = 2, const int = 9,
+        const float = 180.0f, const int = 1);
+    virtual cv::Mat computeHOG(const cv::Mat &);
+    template<typename T>
+    T computeHOGHistogramDistances(
+        const cv::Mat &, std::vector<cv::Mat> &,
+        const int = CV_COMP_BHATTACHARYYA);
+    cv::Mat visualizeHOG(
+        cv::Mat &, cv::Mat &, cv::Size, int, double);
```

Comment:
very hard to see which function did you added and and removed. did you add visualizeHOG() and update arguments of constructor? rename bilinearBinVoting to histgramBinVoting?